TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/support/SocketConnectionObservable.java

```diff
@@ -4,60 +4,65 @@
 
 import com.novoda.tpbot.Result;
 
-import io.socket.client.Ack;
 import io.socket.client.Socket;
 import io.socket.client.SocketIOException;
 import io.socket.emitter.Emitter;
 
-import static io.socket.client.Socket.EVENT_CONNECT_ERROR;
+import static io.socket.client.Socket.*;
 
 public class SocketConnectionObservable extends Observable<Result> {
 
     private final Socket socket;
     private final Handler handler;
-    private final Event event;
 
-    public SocketConnectionObservable(Socket socket, Handler handler, Event event) {
+    public SocketConnectionObservable(Socket socket, Handler handler) {
         this.socket = socket;
         this.handler = handler;
-        this.event = event;
     }
 
     @Override
     public Observable<Result> start() {
-        if (socket.connected()) {
-            notifyOf(Result.from("Already connected"));
-            return this;
-        }
-
-        socket.emit(event.rawEvent(), "", joinSuccessAcknowledgementListener);
         socket.on(EVENT_CONNECT_ERROR, connectionTimeoutListener);
+        socket.on(EVENT_ERROR, connectionErrorListener);
+        socket.on(EVENT_CONNECT, connectionEstablishedListener);
         socket.connect();
         return this;
     }
 
-    private final Ack joinSuccessAcknowledgementListener = new Ack() {
+    private final Emitter.Listener connectionTimeoutListener = new Emitter.Listener() {
```

Comment:
Remove custom `connection` event in favour of using a basic listener that is tied to already built-in socket io events.